repose-aggregator/components/filters/tenant-culling-filter/src/main/java/org/openrepose/filters/tenantculling/TenantCullingFilter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.tenantculling;
+
+import org.openrepose.commons.utils.io.ObjectSerializer;
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper;
+import org.openrepose.core.services.datastore.Datastore;
+import org.openrepose.core.services.datastore.DatastoreService;
+import org.openrepose.filters.keystonev2.KeystoneRequestHandler;
+import org.openrepose.filters.keystonev2.KeystoneV2Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.collection.JavaConverters;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+import static javax.servlet.http.HttpServletResponse.SC_UNAUTHORIZED;
+import static org.openrepose.commons.utils.http.OpenStackServiceHeader.TENANT_ID;
+
+/**
+ * Created by adrian on 6/12/17.
+ */
+@Named
+public class TenantCullingFilter implements Filter {
```

Comment:
is it worth redoing this in scala since we are going to own it now? I can't imagine that would take long.

Reply:
Maybe leave the test in it's current java form to prove the new version does what the old did.

Reply:
Filter converted; test not.